pymeasure/instruments/aja/dcxs.py

```diff
@@ -0,0 +1,229 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import pyvisa
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+
+def truncated_string(value, maxlength):
+    """ Provides a validator function that truncates a string to a maximum length.
+    It returns the original value if its length is smaller or equal to the maximum,
+    otherwise only the characters up to the maximum length will be returned.
+
+    :param value: a value to test
+    :param maxlength: maximum length of the string ("values" arg of the property factories)
+    """
+    if len(value) <= maxlength:
+        return value
+    else:
+        return value[:maxlength]
```

Comment:
You can make it even shorter:
```suggestion
    return value[:maxlength]
```
A slice returns up to that many elements.

Reply:
I finally removed the validator and put the single statement in a lambda functions